zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -3031,6 +3031,13 @@ private ClientCnxnSocket getClientCnxnSocket() throws IOException {
         if (clientCnxnSocketName == null) {
             clientCnxnSocketName = ClientCnxnSocketNIO.class.getName();
         }
+
+        if (clientCnxnSocketName.equals(ClientCnxnSocketNetty.class.getSimpleName())) {
```

Comment:
Why are you using getSimpleName ?

Can you please explain how did you test this patch ?
I am not sure it is worth to add an integration test but we should verify that this works well.

Reply:
thx for your review
if jvm set -Dzookeeper.clientCnxnSocket=ClientCnxnSocketNetty or ClientCnxnSocketNIO，
in zookeeper jar，clientCnxnSocketName is org.apache.zookeeper.ClientCnxnSocketNetty  or org.apache.zookeeper.ClientCnxnSocketNIO
and in shaded zookeeper jar, clientCnxnSocketName is ${shaded.prefix}.org.apache.zookeeper.ClientCnxnSocketNetty or ${shaded.prefix}.org.apache.zookeeper.ClientCnxnSocketNIO
clientCnxnSocketName value is always correct.

Reply:
If other components (like hbase, hdfs...) already use the `-Dzookeeper.clientCnxnSocket=org.apache.zookeeper.*`,  this patch has no effect on them.